contract/signals.py

```diff
@@ -88,9 +91,36 @@ def append_contract_filter(sender, **kwargs):
                     )
 
 
+# additional filters for InsureePolicy in contract
+def append_contract_policy_insuree_filter(sender, **kwargs):
+    user = kwargs.get("user", None)
+    additional_filter = kwargs.get('additional_filter', None)
+    if "contract" in additional_filter:
+        # then check perms
+        if user.has_perms(InsureeConfig.gql_query_insuree_policy_perms) or user.has_perms(PolicyholderConfig.gql_query_insuree_policy_portal_perms):
+            contract_id = additional_filter["contract"]
+            contract_to_process = Contract.objects.filter(id=contract_id).first()
+            # check if user is linked to ph in policy holder user table
+            type_user = f"{user}"
+            # related to user object output (i) or (t)
+            # check if we have interactive user from current context
+            if '(i)' in type_user:
+                ph_user = PolicyHolderUser.objects.filter(Q(policy_holder__id=contract_to_process.policy_holder.id, user__id=user.i_user.id)).first()
```

Comment:
I just realized that you are not making sure the access is still valid should add something like
validFrom <= now(), Q(validTo__isnull=True )|Q(validTo>now() )

Reply:
that is true for all signals

Reply:
Ok. I will add this to check this.

Reply:
@delcroip I added that condition to check validity of that PHUser. Later in OFS-284 I will add (probably tomorrow) for another signals in PH and Contract for Payment  query processing.